tools/toolchains/arm.py

```diff
@@ -70,7 +72,13 @@ def __init__(self, target, notify=None, macros=None,
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
 
-        self.check_c_lib_supported(target, "arm")
+        toolchain = "arm"
```

Comment:
Having to specify this looks wrong, since you are already 'in' the ARM toolchain class.

Reply:
I am not sure if the toolchain name is available nor how to obtain it. It was previously already specify as such I just put it in a variable as I use it more than once.

Reply:
Where are the toolchain instantiated?

Reply:
The toolchain may well be instantiate in more than one place. My only assertion was that the passing the ID of the toolchain to a function inside a class which is specific to that toolchain seems unnecessary.

Reply:
The base class needs to know the toolchain name in order to perform the check. it is necessary.

Reply:
Ok.